cidc_api/models/templates/sync_schemas.py

```diff
@@ -0,0 +1,226 @@
+    """
+    errors = []
+    for event_name in trial.metadata_json["allowed_collection_event_names"]:
+        try:
+            session.merge(
+                CollectionEvent(
+                    event_name=event_name, trial_id=new_trial.protocol_identifier
+                )
+            )
+        except Exception as e:
+            errors.append(e)
+
+    return errors
+
+
+def _sync_cohorts(
+    trial: TrialMetadata, new_trial: ClinicalTrial, session: Session
+) -> List[Exception]:
+    """
+    Given corresponding TrialMetadata and ClinicalTrial instances, copy over cohorts.
+    In JSON, stored as list[string] at /allowed_cohort_names.
+    """
+    errors = []
+    for cohort_name in trial.metadata_json["allowed_cohort_names"]:
+        try:
+            session.merge(
+                Cohort(cohort_name=cohort_name, trial_id=new_trial.protocol_identifier)
+            )
+        except Exception as e:
+            errors.append(e)
+
+    return errors
+
+
+def _sync_shipments(
+    trial: TrialMetadata, new_trial: ClinicalTrial, session: Session
+) -> List[Exception]:
+    """
+    Given corresponding TrialMetadata and ClinicalTrial instances, copy over shipments.
+    In JSON, stored as list[shipment] at /shipments, where each shipment is dict[<column name> : <value>]
+    """
+    errors = []
+    for shipment in trial.metadata_json.get("shipments", []):
+        try:
+            session.merge(
+                Shipment(
+                    trial_id=new_trial.protocol_identifier,
+                    **{
+                        c.name: shipment[c.name]
+                        for c in Shipment.__table__.columns
+                        if c.name in shipment
+                    },
+                )
+            )
+        except Exception as e:
+            errors.append(e)
+
+    return errors
+
+
+def _sync_participants_and_samples(
+    trial: TrialMetadata, new_trial: ClinicalTrial, session: Session
+) -> List[Exception]:
+    """
+    Given corresponding TrialMetadata and ClinicalTrial instances, copy over participants and samples.
+    In JSON, stored as list[participant] at /participants, where each participant is dict[<column name> : <value>]
+    with /participants/*/samples : list[sample] where each sample is dict[<column name> : <value>].
+
+    For Participant, JSON participant_id becomes relational trial_participant_id for clarity.
+    """
+    errors = []
+    shipment_map = _make_sample_to_shipment_map(trial.trial_id, session)
+
+    participants = trial.metadata_json["participants"]
+    for partic in participants:
+        # special handling for name change
+        partic["trial_participant_id"] = partic.pop("participant_id")
+
+        try:
+            # need to keep this for samples
+            new_partic = session.merge(
+                Participant(
+                    trial_id=new_trial.protocol_identifier,
+                    **{
```

Comment:
Just a stylistic comment: I think I understand this argument expansion here but it might be helpful to comment this block of code explaining what it is doing, or perhaps even converting it to a function with a docstring for future explainability. I'll defer to you on the final call, not a deal breaker